### PR TITLE
Improved job serializer

### PR DESCRIPTION
**Pull request by haimasree**
Closes: #1164

### CODE REVIEW

**Comment by haimasree** [comment]
Yes I understand but where do I add this test? I would like to put this with api tests in algorithm app. Can you point me to where api functionalities for algorithm app are tested?

**Comment by jmsmkn** [comment]
Make a new file and do it there.

File: app/tests/algorithms_tests/test_api.py [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
+from tests.algorithms_tests.factories import AlgorithmResultFactory
+from tests.factories import UserFactory
+from tests.utils import get_view_for_user
+
+
+@pytest.mark.django_db
+def test_job_list(client):
+    user = UserFactory()
+    result = AlgorithmResultFactory(job__creator=user)
+    job = Job.objects.get(pk=result.job.pk)
```
Comment (jmsmkn) [inline app/tests/algorithms_tests/test_api.py]:
```suggestion
    job = result.job
```

File: app/tests/algorithms_tests/test_api.py [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
+    response = get_view_for_user(
+        viewname="api:algorithms-job-detail",
+        client=client,
+        user=user,
+        reverse_kwargs={"pk": result.job.pk},
+        content_type="application/json",
+    )
+    assert response.status_code == 200
+    assert job.status == 0
```
Comment (jmsmkn) [inline app/tests/algorithms_tests/test_api.py]:
```suggestion
    assert job.status == job.PENDING
```

File: app/tests/algorithms_tests/test_api.py [hunk 3]
```diff
@@ -1,31 +1,29 @@
...
-    assert response.json()["results"][0]["status"] == "Queued"
-    assert (
-        urlparse(response.json()["results"][0]["result"]).path
-        == urlparse(r.api_url).path
-    )
+from urllib.parse import urlparse
+
+import pytest
+
+from grandchallenge.algorithms.models import Job
```
Comment (jmsmkn) [inline app/tests/algorithms_tests/test_api.py]:
```suggestion
```

**Pull request merged**